Client/Frontend/Home/HistoryPanel.swift

```diff
@@ -23,52 +23,64 @@ private typealias CategoryNumber = Int
+class HistoryPanel: UIViewController, HomePanel {
     weak var homePanelDelegate: HomePanelDelegate? = nil
-    private lazy var emptyStateOverlayView: UIView = self.createEmptyStateOverview()
-
-    private let QueryLimit = 100
-    private let NumSections = 4
-    private let Today = getDate(dayOffset: 0)
-    private let Yesterday = getDate(dayOffset: -1)
-    private let ThisWeek = getDate(dayOffset: -7)
-
-    // Category number (index) -> (UI section, row count, cursor offset).
-    private var categories: [CategorySpec] = [CategorySpec]()
-
-    // Reverse lookup from UI section to data category.
-    private var sectionLookup = [SectionNumber: CategoryNumber]()
-
-    var refreshControl: UIRefreshControl?
+    var profile: Profile!
+    private lazy var tableViewController: HistoryPanelViewController = {
+        return HistoryPanelViewController()
+    }()
+    private lazy var syncedTabsHeaderCell: SyncedTabsTableViewCell = {
+        return SyncedTabsTableViewCell()
+    }()
 
     init() {
         super.init(nibName: nil, bundle: nil)
         NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(HistoryPanel.notificationReceived(_:)), name: NotificationFirefoxAccountChanged, object: nil)
         NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(HistoryPanel.notificationReceived(_:)), name: NotificationPrivateDataClearedHistory, object: nil)
         NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(HistoryPanel.notificationReceived(_:)), name: NotificationDynamicFontChanged, object: nil)
+
+        self.tableViewController.homePanelDelegate = homePanelDelegate
     }
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        self.tableView.accessibilityIdentifier = "History List"
-    }
 
-    override func viewWillAppear(animated: Bool) {
-        super.viewWillAppear(animated)
+        self.tableViewController.tableView.accessibilityIdentifier = "History List"
+        tableViewController.profile = self.profile
+        tableViewController.homePanelDelegate = homePanelDelegate
+        tableViewController.historyPanel = self
 
-        // Add a refresh control if the user is logged in and the control was not added before. If the user is not
-        // logged in, remove any existing control but only when it is not currently refreshing. Otherwise, wait for
-        // the refresh to finish before removing the control.
-        if profile.hasSyncableAccount() && self.refreshControl == nil {
-            addRefreshControl()
-        } else if self.refreshControl?.refreshing == false {
-            removeRefreshControl()
+        syncedTabsHeaderCell.homePanelDelegate = homePanelDelegate
+        syncedTabsHeaderCell.profile = profile
+        syncedTabsHeaderCell.historyPanelController = tableViewController
+        syncedTabsHeaderCell.navigationController = self.navigationController
+
+        self.addChildViewController(tableViewController)
```

Comment:
Whenever adding child view controllers, make sure to also call `didMoveToParentViewController`. https://developer.apple.com/library/ios/featuredarticles/ViewControllerPGforiPhoneOS/ImplementingaContainerViewController.html has some additional documentation for the best practices when adding/removing child view controllers.